apollo-federation/src/sources/source/federated_query_graph/builder.rs

```diff
@@ -26,32 +26,70 @@ pub(crate) trait FederatedQueryGraphBuilderApi {
 }
 
 pub(crate) struct FederatedQueryGraphBuilders {
-    builders: IndexMap<SourceKind, FederatedQueryGraphBuilder>,
+    graphql: FederatedQueryGraphBuilder, // Always the GraphQL variant
+    connect: FederatedQueryGraphBuilder, // Always the Connect variant
 }
 
 impl FederatedQueryGraphBuilders {
     pub(crate) fn process_subgraph_schemas(
         &self,
-        _subgraphs: ValidFederationSubgraphs,
-        _builder: &mut IntraSourceQueryGraphBuilder,
+        subgraphs: ValidFederationSubgraphs,
+        builder: &mut IntraSourceQueryGraphBuilder,
     ) -> Result<(), FederationError> {
-        todo!()
+        subgraphs.into_iter().try_for_each(|(_name, graph)| {
+            let src_kind = extract_source_kind(&graph);
+            builder.set_source_kind(src_kind);
+            self.get_builder(src_kind)
+                .process_subgraph_schema(graph, builder)
+        })
+    }
```

Comment:
can this be tested in any way?

Reply:
The main invariant that I think this needs to hold is that the `GraphQL` and `Connect` `process_subgraph_schema` methods expect to be passed only their respective versions of the `FederatedQueryGraphBuilder`. This is transitively tested when we test both of those implementations. Should they receive an incorrect builder. Is there something else that needs to be tested for?

Reply:
I could add `debug_assert` to make sure the variants line up.